### PR TITLE
🎉 New Source - Sonar Cloud [low-code cdk]

### PR DESCRIPTION
**Pull request by haithem-souala**
## What
A new low code source connector Sonar Cloud

## How
https://sonarcloud.io/web_api

## Streams
* components
* issues
* metrics

## 🚨 User Impact 🚨
N/A

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

Capture d’écran 2022-10-26 à 14 12 00

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      field_pointer: []
+  components_selector:
+    extractor:
+      field_pointer: ["components"]
+  issues_selector:
+    extractor:
+      field_pointer: ["issues"]
+  metrics_selector:
+    extractor:
+      field_pointer: ["metrics"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml:15]: you can have an extractor like:
```yaml
  selector:
    extractor:
      field_pointer:
        - "{{ options['name'] }}"

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  requester:
+    url_base: "https://sonarcloud.io/api"
+    http_method: "GET"
+    request_options_provider:
+      request_parameters:
+        organization: "{{ config['organization'] }}"
+        createdAfter: "{{ config['start_date'] }}"
+        createdBefore: "{{ config['end_date'] }}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml:23]:
```suggestion
        createdBefore: "{{ config['end_date'] or now_utc() }}"
```
Check macros here: https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      $ref: "*ref(definitions.increment_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  metrics_retriever:
+    record_selector:
+      $ref: "*ref(definitions.metrics_selector)"
+    paginator:
+      $ref: "*ref(definitions.increment_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml:66]: You can simplify this after having only one extractor.

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      description: Comma-separated list of component keys.
+      examples:
+        - airbyte-ws-order,airbyte-ws-checkout
+    start_date:
+      title: Start date
+      type: string
+      description: To retrieve issues created after the given date (inclusive).
+      pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}$
+      examples:
+        - YYYY-MM-DD
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:40]: Add a default or make this required. What happens when the request don't use this parameter?

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        Organization key. See <a
+        href="https://docs.sonarcloud.io/appendices/project-information/#project-and-organization-keys">here</a>.
+      examples:
+        - airbyte
+    component_keys:
+      title:  Component Keys
+      type: string
+      description: Comma-separated list of component keys.
+      examples:
+        - airbyte-ws-order,airbyte-ws-checkout
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:33]: Change this to type array

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments, look good!

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml (hunk 1 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml]: Nice tips, done!

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml (hunk 2 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml:15]: That will not work for this case, the api waiting date in specific format: YYYY-MM-DD, the macro return date-time format:
2022-10-26+21%3A02%3A17.320564%2B00%3A00

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml (hunk 3 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml]: Done!

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml (hunk 5 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:33]: Switching to array broke the connector, the api waiting for something like this: componentKeys=component-one,component2
However, the array return this: componentKeys=%5B'component-one',%20'component2'%

I will keep string for now.

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml (hunk 4 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:40]: Nothing, the field will be ignored.

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml (hunk 5 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml]: Fixed.

**Review by marcosmarxm — APPROVED** [review]
Test and Publish commands were executed in https://github.com/airbytehq/airbyte/pull/18584

**Pull request merged**